frontend/server/src/Controllers/User.php

```diff
@@ -3256,6 +3256,101 @@ public static function apiRemoveRole(\OmegaUp\Request $r): array {
         ];
     }
 
+    /**
+     * @return array{token: string}
+     *
+     * @omegaup-request-param null|string $username
+     */
+    public static function apiDeleteRequest(\OmegaUp\Request $r): array {
+        $r->ensureMainUserIdentity();
+        $username = $r->ensureOptionalString(
+            'username',
+            required: false,
+            validator: fn (string $username) => \OmegaUp\Validators::usernameOrEmail(
+                $username
+            )
+        );
+        if (
+            !\OmegaUp\Authorization::isSystemAdmin(
+                $r->identity
+            ) && !is_null(
+                $username
+            )
+        ) {
+            throw new \OmegaUp\Exceptions\ForbiddenAccessException();
+        }
+        $identity = self::resolveTargetIdentity($r);
+        $user = self::resolveTargetUser($r);
+        if (is_null($user)) {
+            throw new \OmegaUp\Exceptions\NotFoundException('userNotExist');
+        }
+        $token = \OmegaUp\SecurityTools::randomString(50);
+        \OmegaUp\DAO\Users::generateDeletionToken($user, $token);
+        self::$log->info(
+            "User {$identity->username} is requesting delete their account."
+        );
+        $email = \OmegaUp\DAO\Emails::getByPK($user->main_email_id);
+        $subject = \OmegaUp\Translations::getInstance()->get(
+            'accountDeletionEmailSubject'
+        );
+        $body = \OmegaUp\ApiUtils::formatString(
+            \OmegaUp\Translations::getInstance()->get(
+                'accountDeletionEmailBody'
+            ),
+            [
+                'verification_id' => strval($username->verification_id),
+            ]
+        );
```

Comment:
You don't need to send the `verification_id` here. It's better send the username:
```suggestion
        $body = \OmegaUp\ApiUtils::formatString(
            \OmegaUp\Translations::getInstance()->get(
                'accountDeletionEmailBody'
            ),
            [
                'username' => $identity->username,
            ]
        );
```

Reply:
oh okay